src/Helpers/OAuth2/OAuthConfig.php

```diff
@@ -56,6 +57,13 @@ class OAuthConfig
      */
     protected string $userEndpoint = 'user';
 
+    /**
+     * Callable that modifies the OAuth requests
+     *
+     * @var callable(\Saloon\Contracts\Request): (void)|null
+     */
+    protected mixed $requestModifier = null;
```

Comment:
I believe it's better to turn this into a Closure, making it more concrete than 'mixed' (anything).
We can still accept callables on methods, but turn them into Closures.
Using the first-class callable syntax on a callable should also bind them to the class, if it's an invokable class.
So there should be no lost information there.

```suggestion
    /**
     * Callable that modifies the OAuth requests
     *
     * @var \Closure(\Saloon\Contracts\Request): (void)|null
     */
    protected Closure $requestModifier = null;
```

Reply:
The reason I don't add a template here, is because it doesn't work, AFAIK.
So we'd need to stick with the raw Request interface as argument.
But we'd need to see what PHPStan feels about once/if everything is applied.

Reply:
Oh awesome, great suggestion thank you!